lib/bulk_processor/job.rb

```diff
@@ -5,9 +5,15 @@ class BulkProcessor
   class Job < ActiveJob::Base
     queue_as 'bulk_processor'
 
-    def perform(records, processor_class, payload)
-      processor = processor_class.constantize.new(records, payload: payload)
-      processor.start
+    def perform(processor_class, payload, file_class, key)
+      file = file_class.constantize.new(key)
+      file.read do |f|
+        csv = CSV.parse(f.read, headers: true)
+        processor = processor_class.constantize.new(csv, payload: payload)
+        processor.start
+      end
+    ensure
+      file.try(:delete)
```

Comment:
Since the job is run asynchronously, the original uploader won't know when this job finishes. Thus this is the only place where clean up makes any sense. The alternative would be to leave the file on s3, but since we're using it as a lock, we should just clean it up.

If we encounter the use case of the original uploader still needing the file, we can add functionality to copy the file and pass the copy to the job (which would remove the copy when done).